np.sh

```diff
@@ -1,8 +1,18 @@
 #!/usr/bin/env sh
+
+if test 0 -ne `git status --porcelain | wc -l`; then
+	echo "Unclean working tree. Commit or stash changes first." >&2;
+	exit 128;
+fi
+
+if test "00" -ne `git fetch; git rev-list --count --left-right @'{u}'...HEAD | awk '{ print $1$2 }'`; then
```

Comment:
Good point on this one. Would you be comfortable with a straight `!=` check for the strings?

Also, I can handle failure of git fetch, but I can handle it in a couple of ways:

Option 1:
```
if test "00" != `git fetch 2> /dev/null && git rev-list --count --left-right @'{u}'...HEAD | awk '{ print $1$2 }'`; then
    echo "There was a problem fetching your git repo or the local/remote history differ. Please push/pull changes." >&2;
   exit 128;
fi
```

Or, option 2:

```
result=$(git fetch 2> /dev/null && git rev-list --count --left-right @'{u}'...HEAD | awk '{ print $1$2 }')
if test -z "$result"; then
    echo "There was a problem fetching your branch" >&2;
    exit 128;
fi
if test "00" != "$result"; then
    echo "Local/Remote history differ. Please push/pull changes." >&2;
    exit 128;
fi
```

Personally I prefer the second, but I'm willing to do either. Or if there's a third way you'd prefer, let me know :)